examples/stm32/f0/other/can/can.c

```diff
@@ -0,0 +1,182 @@
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * Copyright (C) 2010 Thomas Otto <tommi@viadmin.org>
+ * Copyright (C) 2010 Piotr Esden-Tempski <piotr@esden.net>
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <libopencm3/stm32/rcc.h>
+#include <libopencm3/stm32/flash.h>
+#include <libopencm3/stm32/gpio.h>
+#include <libopencm3/cm3/nvic.h>
+#include <libopencm3/cm3/systick.h>
+#include <libopencm3/stm32/can.h>
+
+
+#define TS1FROMBTRVAL(value)	(value & 0xF0000) //((1<<16)|(1<<17)(1<<18)(1<<19)))
+#define TS2FROMBTRVAL(value)	(value & 0x700000)//((1<<20)|(1<<21)(1<<22)))
+#define SJWFROMBTRVAL(value)	(value & 0x3000000)//((1<<24)|(1<<24)))
+#define BRPFROMBTRVAL(value)	((value & 0x1FF)+1)
```

Comment:
This look like they're missing definitions in the library itself?  The bit "explanations" are also not really explaining anything.